README.md

```diff
@@ -2,151 +2,78 @@
+```
 
-In the offchain case, the service provider signs a `Swap` note with the `client` being the `beneficiary`, the `amount` as the security deposit, `SwearSwap` as the `witness` and the service timeout as `validUntil`. If there is no dispute over the service the note will eventually expire and the provider is no longer at risk of losing the deposit. No onchain activity is required in this case (except potentially making sure everything is covered by hard deposits).
+This will generate the bindings in the `bindings/` directory. Suitable versions of `solc` and `abigen` have to be installed for this to work.
+Alternatively this can also be done through docker:
 
-If the `client` is not satisfied with the service a trial can be started using `startTrialFromNote`. In this case `Swear` expects the `remark` field to equal `keccak256(rules, payload)` where `rules` is the address of the trial rules and payload can be arbitrary 32 bytes. `Swear` then records the commitment (with a special `note` flag) and instructs `Swindle` to start the trial (which proceeds in the same way as in the onchain case).
+```sh
+docker build -t sw3 .
+docker run -v $(pwd)/bindings:/sw3/bindings sw3 npm run abigen
+```
 
-If `Swindle` calls compensate no payout takes place, instead `Swear` records that it should return `VALID` if being called as a `Witness` by the `Swap` with the corresponding `noteId` as `payload`.
+In addition to the file from `abigen` this will also generate a go file that includes the runtime bytecode.
 
-Starting multiple trials for the same note is currently broken. As hard deposits can currently be decreased to 0 within 2 days all offchain service provisions have de-facto no solvency guarantees.
+## Overview
 
-## Tests
+### SimpleSwap
 
-There are two test trials in this repository: `OracleTrial` and `SimpleTrial` (with `HashWitness`).
+`SimpleSwap` is a chequebook-style contract with support hard deposits.
 
-#### OracleTrial
+#### Cheques
 
-In the `OracleTrial` there a two `OracleWitness`es. An `OracleWitness` is a `Witness` meant for testing whose testimony for some `noteId` can be set in advance. For the `OracleTrial` to reach a `GUILTY` verdict both of the witnesses need to return a `VALID` testimony. In all other cases the trial ends with `NOT_GUILTY`.
+The owner can issue cheques. Those are a signed piece of data containing the swap address, a beneficiary, and a cumulative amount. If the beneficiary wishes to cash the cheque it needs to be sent to the contract using the `cashChequeBeneficiary` function. Alternatively anyone else can also cash the cheque using `cashCheque` provided they have a signature from the beneficiary. In that case they might also get a portion of the payout as a fee. If there is not enough liquid balance in the contract part of the cheque might bounce (whether the full cheque was paid out should be verified through the `ChequeCashed` event). Later the beneficiary can try cashing again to get the remaining amount.
```

Comment:
Owner is not a used variable anymore. I would rephrase to correctly introduce `issuer`

Reply:
fixed